Submitted By: Joe Ciccone <jciccone@gmail.com>
Date: 2010-08-02
Initial Package Version: 2.6.18
Upstream Status: Unknown
Origin: Joe Ciccone 
Description: This patch gives the ability to change the libdir.
             Updated for Iproute2-2.6.28 by Jim Gifford
             Updated for Iproute2-2.6.24 by Joe Ciccone

diff -Naur iproute2-2.6.34.orig/Makefile iproute2-2.6.34/Makefile
--- iproute2-2.6.34.orig/Makefile	2010-05-19 15:32:43.000000000 +0000
+++ iproute2-2.6.34/Makefile	2010-08-03 01:55:53.157660978 +0000
@@ -12,7 +12,7 @@
 
 SHARED_LIBS = y
 
-DEFINES= -DRESOLVE_HOSTNAMES -DLIBDIR=\"$(LIBDIR)\"
+DEFINES= -DRESOLVE_HOSTNAMES -DLIBDIR=\"$(LIBDIR)\" -DARPDDIR=\"$(ARPDDIR)\"
 ifneq ($(SHARED_LIBS),y)
 DEFINES+= -DNO_SHARED_LIBS
 endif
diff -Naur iproute2-2.6.34.orig/include/iptables.h iproute2-2.6.34/include/iptables.h
--- iproute2-2.6.34.orig/include/iptables.h	2010-05-19 15:32:43.000000000 +0000
+++ iproute2-2.6.34/include/iptables.h	2010-08-03 01:56:54.042453134 +0000
@@ -4,8 +4,12 @@
 #include "iptables_common.h"
 #include "libiptc/libiptc.h"
 
+#ifndef LIBDIR
+#define LIBDIR "/usr/lib"
+#endif
+
 #ifndef IPT_LIB_DIR
-#define IPT_LIB_DIR "/usr/local/lib/iptables"
+#define IPT_LIB_DIR LIBDIR "/iptables"
 #endif
 
 #ifndef IPPROTO_SCTP
diff -Naur iproute2-2.6.34.orig/misc/arpd.c iproute2-2.6.34/misc/arpd.c
--- iproute2-2.6.34.orig/misc/arpd.c	2010-05-19 15:32:43.000000000 +0000
+++ iproute2-2.6.34/misc/arpd.c	2010-08-03 01:53:40.570451501 +0000
@@ -40,7 +40,7 @@
 int resolve_hosts;
 
 DB	*dbase;
-char	*dbname = "/var/lib/arpd/arpd.db";
+char	*dbname = ARPDDIR "/arpd.db";
 
 int	ifnum;
 int	*ifvec;
diff -Naur iproute2-2.6.34.orig/netem/Makefile iproute2-2.6.34/netem/Makefile
--- iproute2-2.6.34.orig/netem/Makefile	2010-05-19 15:32:43.000000000 +0000
+++ iproute2-2.6.34/netem/Makefile	2010-08-03 01:52:20.742452152 +0000
@@ -20,9 +20,9 @@
 	$(HOSTCC) $(CCOPTS) -I../include -o $@ $@.c -lm
 
 install: all
-	mkdir -p $(DESTDIR)/lib/tc
+	mkdir -p $(DESTDIR)$(LIBDIR)/tc
 	for i in $(DISTDATA); \
-	do install -m 755 $$i $(DESTDIR)/lib/tc; \
+	do install -m 755 $$i $(DESTDIR)$(LIBDIR)/tc; \
 	done
 
 clean:
